job/status.go

```diff
@@ -91,3 +91,34 @@ func (c *Config) SetFinished(namespace, name string) {
 		c.Log.Error(err, "could not patch status")
 	}
 }
+
+// GroupJobsByStatus groups jobs by the running state
+func GroupJobsByStatus(jobs []k8upv1alpha1.JobObject) (running []k8upv1alpha1.JobObject, failed []k8upv1alpha1.JobObject, successful []k8upv1alpha1.JobObject) {
```

Comment:
Try to avoid duplicate names. On that caller side this would read `job.GroupJobsByStatus`. I would prefer `job.GroupByStatus`. Less words and more on the natural language side.